### DISCUSSION TITLE
URL for playing m3u playlist via button

### DISCUSSION THREAD

**Discussion by msknight** [music-assistant/support#3374] [Q&A] [answered]
I'm making a button to start a playlist. I'm trying all sorts of URLs but the errors are various.
Server version 2.3.4 on a Pi 5 native home assistant.
File share loaded via SMB.

The most "successful" URL I've had so far is...
media-source://filesystem_smb--LBAprYaX/track/0-playlists/Alan Parsons.m3u
...which gives...
ERROR (MainThread) [music_assistant.webserver] Error handling message: player_queues/play_media: 'NoneType' object has no attribute 'get'

Button code is...

show_name: true
show_icon: true
type: button
tap_action:
  action: perform-action
  perform_action: music_assistant.play_media
  target:
    device_id: 28997742cf314ff142f7af99b3c68430
  data:
    media_type: playlist
    media_id: media://filesystem_smb--LBAprYaX/track/0-playlists/Alan Parsons.m3u
entity: input_button.music_button
grid_options:
  columns: 3
  rows: 2
name: Alan Parsons Project
icon: mdi:record-circle

Grateful for any help because I'm out of options.

**Comment by OzGav** [comment]
The URI is incorrect https://music-assistant.io/faq/how-to/#get-the-uri

**Reply by msknight** [reply]
I've tried a few others including smb:// media:// and they don't work. The page doesn't tell me how to get the URL for the playlists which are located on an SMB share. By the time it's running, the only thing showing up in Developer Tools -> States is the track that's playing, and that's also not the URL to the file, but a URL to the file that it's got in the database which is numbers rather than name.

**Reply by OzGav** [reply, marked as answer]
The first line is 

> For playlists, artists, albums and radio you can simply use the name.

**Reply by OzGav** [reply]
Then there is this https://music-assistant.io/faq/how-to/#add-items-to-the-queue-via-a-script-or-automation

or https://music-assistant.io/faq/massplaymedia/

**Reply by msknight** [reply]
> The first line is
> 
> > For playlists, artists, albums and radio you can simply use the name.

I... er... um... honestly didn't expect that to work.. but it did! Thank you, thank you, thank you.
